decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
```

Comment:
- [ ] this method should also provide a message, which will look something like below:
```
def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
```